bin/awsp.sh

```
@@ -77,6 +84,347 @@ USG
+    echo "→ Successfully upgraded to version $_target_version"
+    echo "→ Restart your shell or run: . \"$_install_dir/awsp.sh\""
+
+    return 0
+  }
+
+  _awsp_upgrade() {
+    echo "awsp upgrade"
+    echo "────────────"
+
+    # Detect installation directory
+    _script_path="${BASH_SOURCE[0]:-${(%):-%x}}"
```

Shellcheck (0.11.0)

[warning] In POSIX sh, BASH_SOURCE is undefined.

(SC3028)